api/client/service/update.go

```diff
@@ -157,10 +152,7 @@ func updateService(spec *swarm.ServiceSpec, flags *pflag.FlagSet) error {
 		updateDurationOpt((flagRestartWindow), task.RestartPolicy.Window)
 	}
 
-	if flags.Changed(flagConstraint) {
-		task.Placement = &swarm.Placement{}
-		updateSlice(flagConstraint, &task.Placement.Constraints)
-	}
+	// TODO: Constraints fied in a differe branch
```

Comment:
Fixed in #23773

I didn't want to repeat the fix here (because it depends on other changes), and I was removing `updateSlice()`, so I left this as a TODO for now. It's going to conflict when I go to merge them, so I shouldn't miss it later.